src/c_bindings/c_wrapper.cpp

```diff
@@ -7,40 +7,57 @@
 extern "C" {
     // C wrapper function
     const char* create_discriminant_wrapper(const uint8_t* seed, size_t seed_size, int length) {
-        std::vector<uint8_t> seedVector(seed, seed + seed_size);
-        integer result = CreateDiscriminant(seedVector, length);
+        try {
+            std::vector<uint8_t> seedVector(seed, seed + seed_size);
+            integer result = CreateDiscriminant(seedVector, length);
 
-        // Serialize the 'result' to a string
-        std::string resultStr = result.to_string();
+            // Serialize the 'result' to a string
+            std::string resultStr = result.to_string();
 
-        // Allocate a new C-string to hold the serialized result
-        char* cResultStr = new char[resultStr.length() + 1];
-        std::strcpy(cResultStr, resultStr.c_str());
+            // Allocate a new C-string to hold the serialized result
+            char* cResultStr = new char[resultStr.length() + 1];
+            std::strcpy(cResultStr, resultStr.c_str());
 
-        return cResultStr; // Return the C-string
+            return cResultStr; // Return the C-string
+        } catch (const runtime_error& e) {
+            return NULL;
```

Comment:
nit-pick: as of C++11, there's a better name for this: `nullptr`

Reply:
I changed it to `nullptr`